src/pages/rooms/[id]/questions/[questionId]/index.tsx

```diff
@@ -1,15 +1,85 @@
-import React, { ReactElement } from 'react';
+// Was kommt in den Callback der Submitfunktion?
+import React, { ReactElement, SyntheticEvent, useState } from 'react';
 import { NextPageContext } from 'next';
 import api from '../../../../utils/api';
 import getInitialProps from '../../../../utils/get-initial-props';
-import IQuestion from '../../../../../types/question';
+import IQuestion, { QuestionDBModel } from '../../../../../types/question';
+import FreeAnswer from './FreeAnswer';
+import RangeAnswer from './RangeAnswer';
+import FixAnswer from './FixAnswer';
+import { useRouter } from 'next/router';
+import Answer, { AnswerCreate } from '../../../../../types/answer';
+import Link from 'next/link';
 
 interface Props {
-  question: IQuestion;
+  question: QuestionDBModel;
 }
 
 export default function Question({ question }: Props): ReactElement {
-  return <>{JSON.stringify(question)}</>;
+  const [rangeAnswer, setRangeAnswer] = useState();
+  const [fixAnswer, setFixAnswer] = useState('');
+  const [freeAnswer, setFreeAnswer] = useState('');
+  const format = question.answersFormat;
+  const router = useRouter();
+  const questionId = question.id;
+  const roomId = router.query.id;
+
+  const answerFunc = (): AnswerCreate => {
+    if (question.answersFormat === 'range') {
+      return { rangeAnswer };
+    } else if (question.answersFormat === 'fix') {
+      return { fixAnswer };
+    } else if (question.answersFormat === 'free') {
+      return { freeAnswer };
+    }
+  };
+
+  const onSubmit = (e: SyntheticEvent) => {
+    e.preventDefault();
+    if (rangeAnswer !== undefined || fixAnswer !== '' || freeAnswer !== '') {
+      api<Answer>(
+        'post',
+        `api/rooms/${roomId}/questions/${questionId}/answers`,
+        () => {
+          router.push(`/rooms/${roomId}/questions/${questionId}/answers`);
+        },
+        answerFunc(),
+      );
+    }
+  };
+
+  return (
+    <>
+      <h4>Poll No.{question.id}</h4>
+      <p>
+        Created at: <b>{question.createdAt.toLocaleDateString()}</b>
+      </p>
+      <h4>Poll question</h4>
+      <p>{question.text}</p>
+      <h4>Answer</h4>
+      <form className="ui form" onSubmit={onSubmit}>
+        {format === 'range' && <RangeAnswer setRangeAnswer={setRangeAnswer} />}
+        {format === 'fix' && (
+          <FixAnswer
+            setFixAnswer={setFixAnswer}
+            fixAnswers={JSON.parse(question.fixAnswers)}
+          />
+        )}
+        {format === 'free' && <FreeAnswer setFreeAnswer={setFreeAnswer} />}
+        <div>
+          <button className="ui button green">Submit</button>
+        </div>
+        <div>
```

Comment:
I think the `div` is not needed. The `Link` and `button` should go out of the `form`. Otherwise button needs `type="button"` attr